File: forminputjava/src/main/java/com/omarshehe/forminputjava/Utils/PasswordStrength.java
package com.omarshehe.forminputjava.Utils;

import android.graphics.Color;

public enum PasswordStrength {
        Weak(0, Color.RED), Medium(1, Color.argb(255, 220, 185, 0)), Strong(2, Color.parseColor("#28A645"));
        static int REQUIRED_LENGTH = 6;
        static int MAXIMUM_LENGTH = 6;
        static boolean REQUIRE_SPECIAL_CHARACTERS = true;
        static boolean REQUIRE_DIGITS = true;
        static boolean REQUIRE_LOWER_CASE = true;
        static boolean REQUIRE_UPPER_CASE = true;
        int resId;
        int color;
        PasswordStrength(int resId, int color) {
            this.resId = resId;
            this.color = color;
        }

        public int getValue() {
            return resId;
        }
        public int getColor() {
            return color;
        }

        public static PasswordStrength calculateStrength(String password) {
            int currentScore = 0;
            boolean sawUpper = false;
            boolean sawLower = false;
            boolean sawDigit = false;
            boolean sawSpecial = false;

            for (int i = 0; i < password.length(); i++) {
                char c = password.charAt(i);

                if (!sawSpecial && !Character.isLetterOrDigit(c)) {
                    currentScore += 1;
                    sawSpecial = true;
                } else {
                    if (!sawDigit && Character.isDigit(c)) {
                        currentScore += 1;
                        sawDigit = true;
                    } else {
                        if (!sawUpper || !sawLower) {
                            if (Character.isUpperCase(c))
                                sawUpper = true;
                            else
                                sawLower = true;
                            if (sawUpper && sawLower)
                                currentScore += 1;
                        }
                    }
                }
            }

            if (password.length() > REQUIRED_LENGTH) {
                if ((REQUIRE_SPECIAL_CHARACTERS && !sawSpecial) || (REQUIRE_UPPER_CASE && !sawUpper) || (REQUIRE_LOWER_CASE && !sawLower) || (REQUIRE_DIGITS && !sawDigit)) {
                    currentScore = 1;
                } else {
                    currentScore = 2;
                }
            } else {
                currentScore = 0;
            }

            switch (currentScore) {
                case 0:
                    return Weak;
                case 1:
                    return Medium;
                case 2:
                    return Strong;
                default:
            }

            return Strong;
        }


}

File: forminputkotlin/src/main/java/com/omarshehe/forminputkotlin/utils/PasswordStrength.kt
package com.omarshehe.forminputkotlin.utils

import com.omarshehe.forminputkotlin.R

class  PasswordStrength {
    enum class PassLevel(resId: Int, color: Int) {
        Weak(0, R.color.colorOnError),Fair(2, R.color.colorOrange), Good(3, R.color.colorBlue), Strong(4, R.color.colorGreen);
        private var mResId: Int = resId
        private var mColor: Int = color

        fun getValue(): Int {
            return mResId
        }

        fun getColor(): Int {
            return mColor
        }

    }
    fun calculateStrength(maxLength:Int,input: String): ArrayList<Any> {
        var result =0
        var upperCasePresent = false
        var specialPresent = false
        var numberPresent = false
        var lengthPresent = false


        for (element in input) {
            when {
                Character.isUpperCase(element) -> upperCasePresent = true
                !Character.isLetterOrDigit(element) && !Character.isWhitespace(element) -> specialPresent = true
                Character.isDigit(element) -> numberPresent = true
                input.length >= maxLength -> lengthPresent = true
            }
        }

        if (upperCasePresent) ++result
        if (specialPresent) ++result
        if (numberPresent) ++result
        if (lengthPresent) ++result

        val level= when (result) {
            0 -> PassLevel.Weak
            1 -> PassLevel.Weak
            2 -> PassLevel.Fair
            3 -> PassLevel.Good
            else -> PassLevel.Strong
        }
        val resultArray=ArrayList<Any>()
        resultArray.add(level)
        resultArray.add(result)
        resultArray.add(upperCasePresent)
        resultArray.add(specialPresent)
        resultArray.add(numberPresent)
        resultArray.add(lengthPresent)
        return resultArray

    }

}